brontide/noise.go

```diff
@@ -289,8 +285,8 @@ type handshakeState struct {
 
 	initiator bool
 
-	localStatic    *btcec.PrivateKey
-	localEphemeral *btcec.PrivateKey
+	localStatic    keychain.SingleKeyECDH
+	localEphemeral keychain.SingleKeyECDH // nolint (false positive)
```

Comment:
Just curious, what was the false lint warning?

Reply:
The linter thinks it's unused because `handshakeState` is embedded into `Machine` and the `localEphemeral` is only ever accessed through the `Machine` struct.
There is an issue still open in the `golangci-lint` repo about false positives for embedded structs: https://github.com/golangci/golangci-lint/issues/826